### ISSUE TITLE
Support Gradle 2.14-rc-1

### ISSUE THREAD

**Issue by johnrengelman** [johnrengelman/shadow#217]
Broken in 2.14-rc-1

```
Caused by: java.lang.ClassCastException: org.gradle.api.plugins.JavaPluginConvention cannot be cast to groovy.lang.GroovyObject
	at com.github.jengelman.gradle.plugins.shadow.ShadowJavaPlugin.configureShadowTask(ShadowJavaPlugin.groovy:57)
	at com.github.jengelman.gradle.plugins.shadow.ShadowJavaPlugin.apply(ShadowJavaPlugin.groovy:35)
	at com.github.jengelman.gradle.plugins.shadow.ShadowJavaPlugin.apply(ShadowJavaPlugin.groovy)
	at org.gradle.api.internal.plugins.ImperativeOnlyPluginApplicator.applyImperative(ImperativeOnlyPluginApplicator.java:35)
	at org.gradle.api.internal.plugins.RuleBasedPluginApplicator.applyImperative(RuleBasedPluginApplicator.java:43)
	at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:137)
```

**Comment by johnrengelman** [comment]
This is due to a breaking change in the Gradle API - https://issues.gradle.org/browse/GRADLE-3455

**Comment by johnrengelman** [comment]
Even if we get past this, there is still an incompatibility:
```
* Exception is:
java.lang.AbstractMethodError: com.github.jengelman.gradle.plugins.shadow.tasks.DefaultInheritManifest.writeTo(Ljava/io/OutputStream;)Lorg/gradle/api/java/archives/Manifest;
	at org.gradle.jvm.tasks.Jar$1$1.execute(Jar.java:65)
	at org.gradle.jvm.tasks.Jar$1$1.execute(Jar.java:59)
```

**Comment by dejan2609** [comment]
@johnrengelman: not sure do you want to go that way or you opt for a different solution... please take some time and review my pull request.

**Comment by dejan2609** [comment]
Basically, solution just mimics changes on Gradle side: https://github.com/gradle/gradle/commit/fb242bbe2d34bc06deab8cfb1e88db07cb038559 (_**Manifests are now written encoded using UTF-8, fix GRADLE-3374**_).

**Comment by dejan2609** [comment]
Update: needs some polishing (tests are failing due to apache commons import issues); see pull request for more details.

**Comment by johnrengelman** [comment]
@dejan2609 thank you for the work of putting together a PR, but I already have changes locally to support this.

**Comment by johnrengelman** [comment]
I'm going to hold a release off until `2.14-rc-2` is released due to the binary incompatibility.

**Comment by dejan2609** [comment]
@johnrengelman ok, I'm going to close my pull request then.

**Comment by eskatos** [comment]
Hi @johnrengelman, we fixed the API change in 2.14-rc-2, since then 2.14-rc-3 is available.
Can you confirm that shadow won't need any change to work with 2.14?
See https://discuss.gradle.org/t/gradle-2-14-rc-3-is-now-available-for-testing/17823

**Comment by johnrengelman** [comment]
Yup - it works
https://github.com/johnrengelman/shadow/commit/bbd1fac6d4f43328f83389b1e2f9be019997098d
